packages/kbn-unified-data-table/src/types.ts

```diff
@@ -19,6 +19,12 @@ import type { DatatableColumnMeta } from '@kbn/expressions-plugin/common';
  */
 export interface UnifiedDataTableSettings {
   columns?: Record<string, UnifiedDataTableSettingsColumn>;
+  /**
+  Optional props passed to Columns to display provided labels as column names instead of field names.
+  This object maps column field names to their corresponding display labels.
+  These labels will take precedence over the data view field names.
+  */
+  columnHeaders?: Record<string, string>;
 }
```

Comment:
Since we moved the option to the settings, it makes more sense if we move the column display text to the already existing column object (UnifiedDataTableSettingsColumn):

```ts
export interface UnifiedDataTableSettings {
  columns?: Record<string, UnifiedDataTableSettingsColumn>;
}

export interface UnifiedDataTableSettingsColumn {
  width?: number;
  /**
  Optional props to display provided string as column name instead of field names.
  This property will take precedence over the data view field names.
  */
	display?: string;
}
```

This way is easier for the consumer as it allows them to configure the settings like this:

```js
const settings = {
    'resource.id' : {
       width: 100,
       display: 'Resource ID'
    }
}
```

Instead of this:

```js
const settings = {
    'resource.id' : {
        width: 100
    },
    columnHeaders : {
        'resource.id' : 'Resource ID'
    }
}
```

Reply:
++ It will keep the API simpler, and with `display` being optional, we won't have to worry about affecting existing usages. With this change the PR should be good to go on the Data Discovery side pending a quick test locally to validate.